content/blog/get-started-with-application-performance-monitoring-tools-overview.md

```diff
@@ -0,0 +1,234 @@
+* provide message queue performance and consuming latency monitoring;
+* start tracing from browser for browser performance monitoring;
+* support infrastructure monitoring for Kubernetes and Linux;
+* support alerting using rules in both observability analysis language and metric analysis language;
+
+Apache SkyWalking provides a list of agents to be used for building *Java*, *.NET Core*, *PHP*, *Node.js*, *Golang*, *LUA*, *Rust* and *C++* apps. It supports to integrate and collect data from multiple sources, including _Prometheus_, _OpenTelemetry_ and _Zabbix_ for metrics and logs, _Zipkin_ for traces. It provides tracing, metrics analysis, alerting, service mesh observability and visualization. 
+
+Apache SkyWalking is an open source tool with *21K* GitHub stars and *6K* GitHub forks. In [StackShare community](https://stackshare.io/apache-skywalking#stacks.), Apache SkyWalking does not yet have many approval, only being  mentioned in 12 developer stacks. However, Apache SkyWalking has more than *600* contributors on GitHub and thousands of contributions every year. All the agents for application instrumentation have been actively maintained. 
+
+Apache SkyWalking is the first open source project that initialized and implemented an [Envoy Access Log Service (ALS)](https://www.envoyproxy.io/docs/envoy/v1.18.2/api-v2/service/accesslog/v2/als.proto) based solution to provide observability on the service mesh, no matter the architecture or language. Since service mesh provides full control of _RPC_, this observation solution is much easier to be added without language specific technology. With this solution, users could get the application service topology map, metrics graph, request details and error message with a very nice visualization. This integration solution can be extremely important for monitoring and visualizing application that consists of many microservices running across on-premises, cloud-based or hybrid environments. 
+
+Apache SkyWalking is lightweight, scalable, and supports alerting and visualization. It can be easily set up as a *self-managed* APM tool within an on-premises data center. This avoids leasing customer data to third party services and removes the restricted security restriction in user environment. 
+
+## Conclusion
+
+This blog post gave a closer look at some of the best APM tools, either open-source or commercial vendor based. It listed the key features of each APM tool, and discussed in details their strengths and weaknesses. The importance of a good APM solution is now indisputable. All it takes is to pick the right one based on your monitoring needs for your applications.   
+
+This blog post is the first of three in a series. In the following posts, I will show you the detailed process how to set up the *Apache SkyWalking*, as the APM tool, in HPE GreenLake for Private Cloud Enterprise for monitoring and alerting both customer application and Kubernetes cluster. 
```

Comment:
Hi Guoping, now that I better understand what are the blog posts that are part of the series, how about this suggestion for edits:

"This blog post is the first of three in a series. In the following post, I will show you the detailed process on [how to set up the ***Apache SkyWalking*** APM tool in HPE GreenLake for Private Cloud Enterprise](https://developer.hpe.com/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud/) for monitoring and alerting both customer application and Kubernetes cluster. Finally, I will expand on using Apache SkyWalking APM tool [to monitor the ***Online Boutique*** microservices application deployed across the hybrid cloud environment](https://developer.hpe.com/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh/).

Notice that Dale might have a better idea.

What do you think?
Denis

Reply:
@GuopingJia 

Based on our discussion in the call today, I think one suggestion would be:

This blog post is the first of three in a series. [In the second post of the series](https://developer.hpe.com/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud/), I will show you the detailed process on how to set up the Apache SkyWalking APM tool for monitoring and alerting of customer applications deployed on Kubernetes cluster provisioned on HPE GreenLake for Private Cloud Enterprise. [In the last (or third) post of the series](https://developer.hpe.com/blog/set-up-apache-skywalking-for-k8s-monitoring-in-hpe-greenlake-for-private-cloud-enterprise/), I will expand on setting up Apache SkyWalking APM tool to monitor the infrastructure of Kubernetes clusters deployed on HPE GreenLake for Private Cloud Enterprise.